FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -194,22 +197,44 @@ extension ComposeViewController {
     }
 
     @objc private func handleSendTap() {
-        sendMessage()
+        prepareSigningKey()
     }
 }
 
 // MARK: - Message Sending
 
 extension ComposeViewController {
-    private func sendMessage() {
+    private func prepareSigningKey() {
+        guard let key = try? keyService.getPrvKeyInfo().get().first else {
+            showAlert(message: "No available private key has your user id \"\(email)\" in it. Please import the appropriate private key.")
+            return
+        }
```

Comment:
Thi does not follow the logic I outlined. You're supposed to evaluate all available private keys, and choose the one that has the appropriate OpenPGP user id encoded in it, to match the sending email, as described in the issue.

Reply:
There are things to filter on database level and then also on openpgp-metadata level. On database level, we have already been doing it - filter by the key belonging to the currently signed-in user. That gives you one or more keys (don't just pick .first)

Then, on that list of keys, you need to filter by primary and secondary OpenPGP user ids. You'll have to run `prvKeyInfo.private` through `Core.parseKeys`. Put the resulting `KeyDetails` into one big array. On each KeyDetails, evaluate `keyDetails.users` - that's what you'll be using for filtering based on my above comments. You can add a computed variable `.primaryPgpUserEmail` which is the email address of the first `user` (technically could be nil). And a computed variable `pgpUserEmails` which are email addresses of all `users`. 

This is assuming that `users` contain something like `["Joe <joe@example.com>", "Bill <bill@example.com>"]`. Which will each have to be parsed to `{name?: "Joe", "email": "joe@example.com" }` - probably our IMAP library will have a parser for this.